src/link-manager.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * @fileoverview Link manager for controlling the link query.
+ */
+
+/** @const */
+genotet.linkManager = {};
+
+/**
+ * Object that stores the link views by their type.
+ */
+/** @type {!Object<!Array<{
+ *    target: genotet.View,
+ *    action: string,
+ *    response: string
+ *  }>>}
+ */
+genotet.linkManager.links = {};
+
+/** @const {!Array<string>} */
+genotet.linkManager.EXPRESSION_ACTIONS = ['addGeneProfile'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.NETWORK_ACTIONS = ['nodeClick', 'edgeClick'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.BINDING_ACTIONS = ['updateTrack', 'locus'];
+
+/** @const {string} */
+genotet.linkManager.DEFAULT_MAPPING_FILENAME = 'namecode';
+
+/**
+ * Initializes the link manager.
+ */
+genotet.linkManager.init = function() {
+  genotet.linkManager.links = {};
+  genotet.data.geneBindingMappingFile =
+    genotet.linkManager.DEFAULT_MAPPING_FILENAME;
+};
+
+
+/**
+ * Set up the link manager.
+ */
+genotet.linkManager.link = function() {
+  for (var linkViewName in genotet.linkManager.links) {
+    var linkView = genotet.viewManager.views[linkViewName];
+    $(linkView)
+      .on('genotet.link', function(event, linkData) {
+        switch (linkData.action) {
+          case 'nodeClick':
+            var node = linkData.data;
+            var genes = [node.id];
+            break;
+          case 'edgeClick':
+            var edge = linkData.data;
+            var genes = edge.id.split(',');
+            break;
+        }
+        genotet.linkManager.links[linkViewName]
```

Comment:
This is bad design you're having too many different types of functionality within the same function. The link manager should just passed whatever message to view handlers. Actual handling should be written in view rather than in the manager. For example link manager should never care about loading mapping files. The views whoever need the mapping file processes that